code/game/objects/items/storage/boxes.dm

```diff
@@ -863,16 +863,82 @@
 	for(var/i in 1 to 6)
 		new /obj/item/ammo_casing/shotgun/beanbag(src)
 
-/obj/item/storage/box/actionfigure
+/obj/item/storage/box/figurines
 	name = "box of action figures"
-	desc = "The latest set of collectable action figures."
+	desc = "The latest set of collectable figurines."
 	icon_state = "box"
 
-/obj/item/storage/box/actionfigure/PopulateContents()
-	for(var/i in 1 to 4)
-		var/randomFigure = pick(subtypesof(/obj/item/toy/figure))
+/obj/item/storage/box/figurines/PopulateContents()
+	for(var/i in 1 to 3)
+		var/randomFigure = pick(
+			pick(subtypesof(/obj/item/toy/figure)),
+			pick(subtypesof(/obj/item/toy/figure)),
+			pick(subtypesof(/obj/item/toy/figure)),
+			pick(subtypesof(/obj/item/toy/prize)),
+			pick(subtypesof(/obj/item/toy/prize)),
+			pick((subtypesof(/obj/item/toy/plush) - typesof(/obj/item/toy/plush/goatplushie/angry/))),
+			pick((subtypesof(/obj/item/toy/talking) - typesof(/obj/item/toy/talking/codex_gigas))))

```

Comment:
Build and cache those lists on a global, then pick from them.

Reply:
I'm a bit confused as to how to do this. The intention here is to weight the categories without having to weight the individual entries. Is there a way to build lists while also assigning them categorical weights? If I type out the individual entries I'm adding like 70+ entries. If I try using the `pick` function in a global list, it activates on init and will always spit out the same pick.

Reply:
`pick()` supports weighted entries: <https://secure.byond.com/docs/ref/info.html#/proc/pick>

So you can do something like this:
```cpp
var/randomFigure = pick(\
3; GLOB.toy_figures,\
2; GLOB.toy_prizes,\
1; GLOB.toy_plushes,\
1; GLOB.toy_talking\
\)
```

Not to have to initialize each list each time you can simply build it once and then make use of it:
```cpp
GLOBAL_LIST_INIT(toy_figures, subtypesof(/obj/item/toy/figure))
GLOBAL_LIST_INIT(toy_plushes, subtypesof(/obj/item/toy/plush) - typesof(/obj/item/toy/plush/goatplushie/angry/))
```
And so on.

Reply:
Ah, someone had told me that `pickweight` had to be used as `pick` wouldn't do weights. Should have just looked at the ref. Thank you @Rohesie !